paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -234,6 +236,10 @@ internal abstract class BaseSheetViewModel(
             isLinkEnabled = linkHandler.isLinkEnabled.asLiveData(),
             initialSelection = savedSelection.asLiveData(),
             isNotPaymentFlow = this is PaymentOptionsViewModel,
+            nameProvider = { code ->
+                val paymentMethod = lpmResourceRepository.getRepository().fromCode(code)
+                application.getString(paymentMethod?.displayNameResource!!)
```

Comment:
Not super happy with this. I can look into a way to make `SupportedPaymentMethod` hold a `displayName` instead of a `displayNameResource`, but that would be the stuff for a follow-up pull request.